pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -23,6 +23,10 @@ const (
 	//
 	// https://kubernetes.io/docs/concepts/services-networking/service/#proxy-protocol-support-on-aws
 	awsLBProxyProtocolAnnotation = "service.beta.kubernetes.io/aws-load-balancer-proxy-protocol"
+	// lbServiceCleanUpFinalizer is a finalizer attached to any service that has type=LoadBalancer.
+	// Upon deletion of the service, the actual deletion of the resource will be blocked until
+	// this finalizer is removed.
+	lbServiceCleanUpFinalizer = "service.kubernetes.io/load-balancer-cleanup"
```

Comment:
We cannot use this in OpenShift 4.2 or earlier because the finalizer is new in Kubernetes 1.16:

> Specifically, if a Service has type LoadBalancer, the service controller will attach a finalizer named service.kubernetes.io/load-balancer-cleanup. The finalizer will only be removed after the load balancer resource is cleaned up. This prevents dangling load balancer resources even in corner cases such as the service controller crashing.
> 
> This feature is beta and enabled by default since Kubernetes v1.16. You can also enable it in v1.15 (alpha) via the feature gate ServiceLoadBalancerFinalizer.

https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#garbage-collecting-load-balancers

Reply:
Can we refactor master to use existence checks before back porting?

Reply:
>Can we refactor master to use existence checks before back porting?

@danehans although I phrased this as a question, I guess it's more of a statement that the refactor is necessary because the information to support the current approach simply does not exist in 4.1 and 4.2 clusters